notebooks/6_Putting_it_All_Together.ipynb

```diff
@@ -0,0 +1,1002 @@
+    "# X_test = X_raw_test[:, idx]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Choosing the active set of environments"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 20,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Selecting 177 samples\n"
+     ]
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "5c838339fa004203a54b93dd3520cdc9",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "HBox(children=(HTML(value=''), FloatProgress(value=0.0, max=177.0), HTML(value='')))"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "from skcosmo.selection import SampleCUR\n",
+    "\n",
+    "num_active = int(len(i_train) / 10)\n",
+    "sample_mixing = 0.5\n",
+    "\n",
+    "print(f\"Selecting {num_active} samples\")\n",
+    "scur = SampleCUR(X=X_train, Y=Yp_train, mixing=sample_mixing, progress_bar=True)\n",
+    "sample_idx = scur.select(num_active)\n",
+    "\n",
+    "X_active = X_train[sample_idx]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# np.savez('active_set_file.npz', idx=sample_idx)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Skip Choosing Active Set"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# # Import active set indices from numpy file\n",
+    "\n",
+    "# index_file = 'active_set_file.npz'\n",
+    "# data = np.load(index_file)\n",
+    "# sample_idx = data['idx']\n",
+    "\n",
+    "# X_active = X_train[sample_idx]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Constructing a Sparse KPCovR"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.metrics.pairwise import pairwise_kernels\n",
+    "from scipy.linalg import svd\n",
+    "\n",
+    "from skcosmo.pcovr import KPCovR\n",
+    "from matplotlib import pyplot as plt\n",
+    "\n",
+    "kernel = \"rbf\"\n",
+    "kernel_params = {\"gamma\": 0.2}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Determine the active kernel\n",
+    "**Keep in mind that this example is originally written with an environment-centred property, and therefore we can use an environment kernel.**\n",
+    "\n",
+    "Until you can specify an active set with sklearn, we'll construct a sparse nystroem approximation in a similar manner, and replace this with the sklearn functionality when it becomes available."
```

Comment:
This sentence is very specific to the time this notebook is written, should we remove it? 

We are not trying to have a notebook that is using sklearn as much as possible.